### PR TITLE
Adapt to possible timewarrior file structures

### LINKED ISSUE #98: Timewarrior seems to change its folder structure
During last installation, timewarrior had a different folder structure than what we assume (`.timewarrior`).

- Config and extensions were put in the `.config/timewarrior` folder
- Data files were put in `./local/share/timewarrior`

Even though this is not documented anywhere, Timewarrior stored the data in the new structure and didn't create a `.timewarrior` folder.
After using Timewsync, which adds the folder and places the returned data into it, Timewarrior started using the classic `.timewarrior` folder again, ignoring the previously-used new structure.

Since users might have data in the new file structure, we might want to update/revise our folder accesses.

### PR DESCRIPTION
**Pull request by arneturuc**
- Adds `paths.py` containing paths and priority based on `paths.cpp` from timewarrior
- Updates usages of paths in `io_handler.py`

Closes #98

We might update our own folder structure similarly in the future

### CODE REVIEW

File: timewsync/paths.py
Comment (vstollen) [inline timewsync/paths.py:1]: ### Non-blocking

Maybe we could add a comment with a reference to [`paths.cpp`](https://github.com/GothenburgBitFactory/timewarrior/blob/e27d91a903cd08af94944c930d8b9a3645d85ff3/src/paths.cpp) for our future selves.

Additionally, I'd prefer to prefix the internal variables (like `uses_legacy_config`, etc.) with an underscore, e.g., `_uses_legacy_config` to signal internal ([PEP 8](https://peps.python.org/pep-0008/#descriptive-naming-styles)).

**Review by vstollen — DISMISSED** [review]
Thanks for your work! Left a comment with two minor suggestions.

**Review by joachimschmidt557 — APPROVED** [review]
LGTM! I agree with @vstollen's points.

**Comment by arneturuc** [comment]
Prefixed the internal variables with `_` and added a docstring to the module.
Feel free to raise additional points, as I would like to test and release the next version on pip after this is merged :)

**Review by joachimschmidt557 — APPROVED** [review]

**Review by vstollen — APPROVED** [review]
Thanks!

**Pull request merged**